### PR TITLE
refactor: (ecam) Convert FCTL page to React

### PR DESCRIPTION
**Pull request by RichardPilbery**
Refactor FCTL page to use Reach and new SimVars. Changes should be invisible to user. (Note was #4193 prior to hard fork).

Fixes #[issue_no]

## Screenshots (if necessary)

ECAM-FCTL-v2
ECAM-FCTL-v2-1

## Additional context

Discord username (if different from GitHub):

## Testing instructions

1. Start engines
2. Check all control service movements (ailerons, elevators, rudder and spoilers) are shown in the ECAM
3. Check spoilers show when armed speedbrake deploys

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by Benjozork** [comment]
@RichardPilbery please resolve conflicts

File: src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx [hunk 1]
```diff
@@ -156,7 +156,6 @@ const WeatherWidget = (props: WeatherWidgetProps) => {
         return Metar.get(icao, source)
             .then((result) => {
                 const metarParse = metarParser(result.metar);
-                console.info(metarParse);
```
Comment (Benjozork) [inline src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx:159]: You don't need to remove those, they are just warnings in ESLint.

**Review by Benjozork — CHANGES_REQUESTED** [review]

File: src/instruments/src/SD/Pages/Fctl/Fctl.tsx [hunk 2]
```diff
@@ -0,0 +1,769 @@
...
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
```
Comment (pepperoni505) [inline src/instruments/src/SD/Pages/Fctl/Fctl.tsx:17]:
```suggestion
```
Our license has changed, and this should be removed

**Review by pepperoni505 — DISMISSED** [review]

File: src/instruments/src/SD/Pages/Fctl/Fctl.tsx (hunk 2 above)
Comment (RichardPilbery) [reply src/instruments/src/SD/Pages/Fctl/Fctl.tsx:17]: Done

**Review by BillyGeoThink2020 — APPROVED** [review]

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -240,6 +240,7 @@
 1. [RMP] Added integral illumination to RMP volume knobs - @ImenesFBW (Imenes)
 1. [RMP] Fixed ACP CALL buttons emissive behavior - @ImenesFBW (Imenes)
 1. [TEXTURE] Fixed decal clipping issues on the overhead fuel panel - @ImenesFBW (Imenes)
+1. [ECAM] Refactor flight control page to use React - @RichardPilbery (tricky_dicky#3571)
```
Comment (pepperoni505) [inline .github/CHANGELOG.md:243]: You should put this at the end of the changelog

File: src/instruments/src/SD/Pages/Fctl/Fctl.tsx (hunk 2 above)
Comment (pepperoni505) [reply src/instruments/src/SD/Pages/Fctl/Fctl.tsx:17]: This is still here

Comment (RichardPilbery) [reply src/instruments/src/SD/Pages/Fctl/Fctl.tsx:17]: That's wierd, I thought I'd used the 'commit suggestion' button. Standby.

File: .github/CHANGELOG.md (hunk 3 above)
Comment (RichardPilbery) [reply .github/CHANGELOG.md:243]: It was when I last updated the changelog...

**Review by Benjozork — APPROVED** [review]

**Comment by starvingjavi** [comment]
Category: QA Tester
Name: Javi#2557
Date of testing: 15/04/2021
Version of the sim: 1.15.7
PR Tested: #4271 
Tier of testing: 1/2/3
Changes to observe:
No problems with FCTL ECAM page during flight

Testing technique: Do a full flight with FCTL page up throughout the whole flight
Steps you've made to test the Pull Request:
1. Start flight at EHAM
2. Program flight to LFPG following the route KUDA1T WOODY N872 CIV Y50 MATIX MATI8W
3. Check for FCTL page and that all flight surfaces are responsive
4. Complete flight

Flight notes: Noneventful. Everything worked as planned


Issues (If Any): None

Overall Rating: Good
Conclusions of the testing: No problems or any issues found during normal operations on the FCTL ECAM page

**Comment by darkfury334** [comment]
Category: QA Trial
Name: DarkFury#4936
Date of testing: Current date 15/04/2021
Version of the sim: 1.15.7.0
PR Tested: #4271
Tier of testing: 2
Changes to observe:
FCTL page Changes

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1. Load C&D at EGKK
2. Start engines
3. Check all control surface movements are correct in lower ecam
4. Take off and do a loop around EGKK at 5000ft
5. overspeed and Arm speedbrakes, check in spoliers show on ecam when activate

Flight notes (When done a full flight):N/A

Media: Insert screenshots, recordings, or anything relevant to the review here.

Issues (If Any):

Overall Rating: Good
Conclusions of the testing: NO issues found

**Comment by TheSquirtingPantin** [comment]
Category: QA Trial
Name: Discord User Rémi#8871
Date of testing: Current date 15/04/2021
Version of the sim: V 1.15.7.0
PR Tested: #4271
Tier of testing: 2
Changes to observe:
Conversion of FCTRL page to react

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1. Full Flight from KPDX to KSFO
2. Visual inspection of FCTRL page during normal and abnormal configurations
3.
4.

Flight notes (When done a full flight):N/A

Media: N/A

Issues (If Any):
1) When FAC2 is deactivated manually, the rudder limits are staying green. (looking for reference to properly document the expected behavior) 


Overall Rating: Good
Conclusions of the testing: LGTM, good job ! Minor logic graphic issue when FAC2 is deactivated.

**Comment by RichardPilbery** [comment]
> 1. When FAC2 is deactivated manually, the rudder limits are staying green. (looking for reference to properly document the expected behavior)

It's likely you'd need to disable both. However, I've asked an engineer who can confirm with the real aircraft.

**Comment by marcman86** [comment]
Category: QA Tester
Name: marcman86#4907
Date of testing: 16.04.2021
Version of the sim: 1.15.7.0
PR Tested: #4271
Tier of testing: 1
Changes to observe: FCTL page works with React

Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned C&D at EDDP
2.) Start AC
3.) Tested functions of the FCTL page
4.) Did a start abort

Issues: 
- When switch from speed brake to speed brake 0 the outer spoiler arrows flashes up
- On start abort with spoilers armed the outer spoiler arrow does not show up (Image1), compared to the wheel page (Image2) 

Media:


Image1


Image2

Overall Rating: To be improved

Conclusions of the testing: Works, but got small bugs with the outer spoiler arrows

Hope this helps :-)

**Comment by FlightSimmer7700** [comment]
Category: QA Tester
Name: FlightSimmer7700#3468
Date of testing: 16/4/21
Version of the sim: 1.15.8.0
PR Tested: #4271 
Tier of testing: 2
Changes to observe:
FCTL page

Testing technique: Completed a full flight
Steps you've made to test the Pull Request:
1. Loaded in and looked at page
2. Turned on engines and looked at page
3. Looked at page in flight
4. Looked at page after landing
5. Looked at page once shut down

Flight notes (When done a full flight):

Media: Insert screenshots, recordings, or anything relevant to the review here.
image
image


Issues (If Any): As mentioned above by marcman86 the outer two and inner 2 don't show as extended in the FCTL page while they do show as extended in the wheels page
Steps to achieve the issue: Extend spoilers and look at both FCTL and Wheel pages

Overall Rating: To be improved
Conclusions of the testing: Everything else but the inner and outer spoilers do not show as extended works

**Comment by RichardPilbery** [comment]
@FlightSimmer7700 When did you take the 1st screenshot? Was this when the speedbrakes were extended? If so, you should only see spoilers 2, 3 and 4 extended. The wheels page can only show all spoilers or no spoilers...it needs to be reworked. On the other hand, all spoilers should deploy when the aircraft lands or aborts takeoff, which does not work as intended, which is the error @marcman86 spotted, which is on my to-do list to fix.

**Comment by FlightSimmer7700** [comment]
@RichardPilbery  This was after landing and I just tried extending them myself to see after I saw the comment by marcman86

**Comment by Benjozork** [comment]
Please remove console.logs and we can merge this

**Pull request merged**